mmdet/core/results/results.py

```diff
@@ -0,0 +1,171 @@
+import copy
+
+import numpy as np
+import torch
+
+
+class Results(object):
+    """Base class of the model's results.
+
+    The model predictions and meta information are stored in this class.
+    User can modify the predictions with `set()` or `remove()`.
+    All predictions results can be export to a dict with `export_results`.
+
+    The meta information(filename,image_size...) about the image are
+    unmodifiable once initialized.
+
+    The keys and values of meta information and predictions
+    can be accessed with `keys()`, `values()`, `items()`.
+
+    All other (non-field) attributes of this class are considered private:
+    they must start with '_'.
+
+    Besides, it support tensor like methods, such as `to()`,`cpu()`,
+    `cuda()`,`numpy()`, which would apply corresponding function
+    to all Tensors in the instance.
+    """
+
+    def __init__(self, img_meta=None, **kwargs):
```

Comment:
1. Does it support the slicing operator?

Reply:
This is a base class of results, we support slicing in subclass `InstanceResults`.